### PR TITLE
Step function graph

### PR DESCRIPTION
**Pull request by JonTheBurley**
This PR adds the Step Function Graph component.

Docs View:
image

Canvas View:
image

### CODE REVIEW

File: .svgarc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+plugins: {
+    removeViewBox: false
```
Comment (JonTheBurley) [inline .svgarc:2]: This was needed for SVGR to not remove the viewBox attribute from SVGs.

File: .svgo.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+plugins:
+  - removeViewBox: false
```
Comment (JonTheBurley) [inline .svgo.yml:2]: This was needed for SVGR to not remove the viewBox attribute from SVGs.

File: src/global.d.ts [hunk 3]
```diff
@@ -4,3 +4,5 @@ declare module '*.svg' {
     const src: any;
     export default src;
 }
+
+declare module '*.js';
```
Comment (JonTheBurley) [inline src/global.d.ts]: This lets us work with imported JS files without needing to type them.

File: src/stories/StepFunctionGraph/stepFunctionGraphData.ts [hunk 4]
```diff
@@ -0,0 +1,573 @@
+export const simpleJson = {
```
Comment (JonTheBurley) [inline src/stories/StepFunctionGraph/stepFunctionGraphData.ts:1]: Check this file for anything that needs to be scrubbed that I may have missed.

Comment (michaelpuzon) [reply src/stories/StepFunctionGraph/stepFunctionGraphData.ts:1]: Data payload looks generic enough.

**Review by michaelpuzon — APPROVED** [review]

**Review by jason-ransom-slalom — APPROVED** [review]
@JonTheBurley - Not going to block, but an issue should be created later if it's not a quick fix. Whenever clicking on a node in canvas, it seems to redraw mostly outside of the window (see screen shots). Otherwise, LGTM 👍🏾 

## Before Click
Screen Shot 2021-04-14 at 7 25 18 AM


## After Click
Screen Shot 2021-04-14 at 7 24 34 AM

**Comment by JonTheBurley** [comment]
> @JonTheBurley - Not going to block, but an issue should be created later if it's not a quick fix. Whenever clicking on a node in canvas, it seems to redraw mostly outside of the window (see screen shots). Otherwise, LGTM 👍🏾

It tries to center on a highlighted node which doesn't mesh super well with Storybook, but I thought it was stable enough in Docs to stay in for demo purposes. Don't think there's anything to do about it aside from removing the click handler from the Storybook stories file.

File: src/StepFunctionGraph/Digraph.ts [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+        // Vertex and Edge counters
+        this.V = 0;
+        this.E = 0;
+
+        // This is the heart of the Digraph, storing outdegree vertices at an index which is the vertex that points to it
+        this.adjacencyMatrix = [];
+        this.vertexDataMap = new Map<number, any>();
+
+        // Indegree represents the vertices that point to a vertex where the index is vertex we want values for
```
Comment (zach-jones-slalom) [inline src/StepFunctionGraph/Digraph.ts:29]: super nitpick. I think "Where the index is vertex" should have a 'the' to read "Where the index is the vertex"

File: src/StepFunctionGraph/Graph.tsx [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+            globalOffset.offset.x = (globalOffset.defaultOffset.x * 2) - xOffset;
+            globalOffset.offset.y = ((globalOffset.defaultOffset.y * 2) - (node.y));
+        }
+    };
+
+    // Construct a Digraph to store a representation of the graph to draw
+    const graph: Digraph = useMemo(() => new Digraph(), [json]);
+
+    // Parse the JSON and return a complete graph
+    // This States field check can be removed once we fit the JSON to an interface
```
Comment (zach-jones-slalom) [inline src/StepFunctionGraph/Graph.tsx:199]: Do you think this should be marked with a TODO or is that not necessary?

**Review by zach-jones-slalom — APPROVED** [review]

File: src/StepFunctionGraph/Graph.tsx (hunk 6 above)
Comment (JonTheBurley) [reply src/StepFunctionGraph/Graph.tsx:199]: I'll just remove the comment about it. I'm not sure an interface will be written for it.

File: src/StepFunctionGraph/Digraph.ts (hunk 5 above)
Comment (JonTheBurley) [reply src/StepFunctionGraph/Digraph.ts:29]: Thanks!

**Pull request merged**